### PR TITLE
Milestone 2: Add reducers and actions

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 2: Add reducers and actions
In this milestone I :
- Added redux to my project
- Setup up redux store
- Implemented Books slice(added addBook and removeBook reducers)
- Implemented Categories slice(added checkStatus reducer)

### CODE REVIEW

File: src/redux/features/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  currentStatus: 'Under Construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => state.currentStatus,
+  },
+});
```
Comment (AmaduKamara) [inline src/redux/features/categories/categoriesSlice.js]: - [ ] Initial state is expected to have categories state to be empty, just as you did, and that state is what you have to check its value if it has the text "Under Construction" and return that else return the state value. 
What if you could implement it this way below?

```javascript
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```

- [ ] You no longer have to have the currentStatus state value. 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @zdnahom,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Used redux toolkit as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/features/categories/categoriesSlice.js (hunk 1 above)
Comment (zdnahom) [reply src/redux/features/categories/categoriesSlice.js]: @AmaduKamara, thanks for the review and suggestions. but the requirement says that the function should always to return "under construction"
image

Comment (zdnahom) [reply src/redux/features/categories/categoriesSlice.js]: Thanks again @AmaduKamara, I will try your method now.

**Review by franclobo — APPROVED** [review]
Hi @zdnahom, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good communication with CR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**